### DISCUSSION TITLE
Segmentation Postprocessing Issues

### DISCUSSION THREAD

**Discussion by cmbean2** [obss/sahi#915] [Q&A]
Hi, I am looking for help with the postprocessing of predictions from my model when using SAHI. My model is a custom Yolov8 instance segmentation model. It is trained on small subsets, but my ultimate application would be to large regions.

I have been able to run the sliced predictions with SAHI and it is giving out generally good predictions (all on features none on background). However, I am getting many overlapping features/predictions that are not being merged together into one prediction and would look for assistance on if this is an issue in SAHI, or if I should simply post-process the predictions more myself, in which case would anyone have intuition on how to go about it. (With the caveat features can cross one another and both masks ideal should consider the overlap, look similar to "X's" in the image).

Thanks - Chris 

The different colors represent different instance segmentation masks.
![image](https://github.com/obss/sahi/assets/118309806/80c39a4e-a829-4b89-9a4c-184e7184a55f)

**Comment by W-hary** [comment]
I also have this problem, have you solved it?

**Reply by cmbean2** [reply]
I have not solved this issue. I have instead shifted to manually coding and postprocessing my masks to perform the same task.

**Comment by filipemendesdev** [comment]
> Hi, I am looking for help with the postprocessing of predictions from my model when using SAHI. My model is a custom Yolov8 instance segmentation model. It is trained on small subsets, but my ultimate application would be to large regions.
> 
> 
> 
> I have been able to run the sliced predictions with SAHI and it is giving out generally good predictions (all on features none on background). However, I am getting many overlapping features/predictions that are not being merged together into one prediction and would look for assistance on if this is an issue in SAHI, or if I should simply post-process the predictions more myself, in which case would anyone have intuition on how to go about it. (With the caveat features can cross one another and both masks ideal should consider the overlap, look similar to "X's" in the image).
> 
> 
> 
> Thanks - Chris 
> 
> 
> 
> The different colors represent different instance segmentation masks.
> 
> ![image](https://github.com/obss/sahi/assets/118309806/80c39a4e-a829-4b89-9a4c-184e7184a55f)
> 
> 

Hey @cmbean2 how did you manage to use it with instance segmentation? I am only getting bounding boxes.

**Reply by cmbean2** [reply]
I don't recall exactly as I have shifted to manually doing the postprocessing, but I basically had to pull the masks and visualize them separately from the automatic output of the model/SAHI which is the bboxes for the visualization.
